### ISSUE TITLE
TKTService: duplicated iteration after immediate stop+start

### ISSUE THREAD

**Issue by tinchodias** [pharo-contributions/taskit#75]
Steps to reproduce:

Start a service that prints the hash of the proocess and a timestamp in Transcript, every second:
~~~Smalltalk
service := TKTParameterizableService new.
service name: UUID new asString.
service stepDelay: 1 second.
service step: [ 
	Processor activeProcess identityHash trace.
	Time now traceCr ].
service start.
~~~

You can see that, effectively, it's printed once per second. Then, do this:
~~~Smalltalk
service stop; start.
~~~

Now you can see it's printed twice per second.

**Comment by tinchodias** [comment]
Now I re-read me report and realize it's wrong because I don't wait for the completion of `stop`. (Also, I didn't realize that `restart`already exists in TKTService, but I don't like the hardcoded 500ms wait... I will create another issue for that). 

Sorry... learning aloud!

**Comment by tinchodias** [comment]
Bah, I re-open it to let @sbragagnolo decide if this is something to take actions or not.

**Comment by tinchodias** [comment]
UPDATE: This bug is reproducible using the valid `restart` method instead of the wrong `stop`+`start` approach I used in my first report, above.
~~~Smalltalk
"start logging to Transcript every second"
service := TKTParameterizableService new.
service name: UUID new asString.
service stepDelay: 1 second.
service step: [ 
	Processor activeProcess identityHash trace.
	Time now traceCr ].
service start.

"reset"
service restart.
~~~
Tested loading current master in a Pharo 9.
